app/main.py

```diff
@@ -1,25 +1,84 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, 0)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    @abstractmethod
+    def validate(self) -> bool:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
+
+    def validate(self) -> bool:
+        return True if self.age and self.height and self.weight else False
```

Comment:
This method is redundant

Reply:
I used this method because I set attribute value to 0 in IntegerRange class if value was out of range, so I needed to catch it later. I couldn't think of any other way. 

I could have used: 
return (self.limitation_class.age == visitor.age and
            self.limitation_class.height == visitor.height and
            self.limitation_class.weight == visitor.weight)
in can_access method directly but it doesn't look nice.

I now replaced it with exception handling, although it's from the following topics as I don't have any other ideas. If there is another way, please let me know.